src/k8sconfiguration/azext_k8sconfiguration/custom.py

```diff
@@ -0,0 +1,176 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+
+from azext_k8sconfiguration.vendored_sdks.models.source_control_configuration_py3 import SourceControlConfiguration
+from azext_k8sconfiguration.vendored_sdks.models.helm_operator_properties import HelmOperatorProperties
+
+
+def show_k8sconfiguration(client, resource_group_name, cluster_name, name, cluster_type='connectedClusters',
+                          configuration_type='sourceControlConfiguration', api_version='2019-11-01-preview'):
+    # Determine ClusterRP
+    if cluster_type.lower() == 'connectedclusters':
+        cluster_rp = 'Microsoft.Kubernetes'
+    elif cluster_type.lower() == 'managedclusters':
+        cluster_rp = 'Microsoft.ContainerService'
+    else:
+        raise CLIError("Invalid cluster-type.  Supported values are 'connectedClusters' and 'managedClusters'.")
```

Comment:
We can define the available value with `get_enum_type` in `load_arguments`.

Reply:
Can you pl. explain?  Can I validate the parameter for one set of values, and also assign corresponding values to another variable, in load_arguments?

Reply:
We have a framework to do such enum type, and don't need to check user input in custom.py. You can reference [here](https://github.com/Azure/azure-cli-extensions/blob/581cb603263d5977c28992eca6c9daea2c3bfe65/src/stream-analytics/azext_stream_analytics/_params.py#L27).

Reply:
Ah, got it.  I am indeed using the get_enum_type for cluster_type parameter.  In this method, I am actually not validating.  I am just determining the value the RP, based on the cluster_type.  I have updated the code to not raise CLIError, as well, since that code will never be reached.